src/java.base/share/classes/sun/security/ssl/SSLSocketImpl.java

```diff
@@ -1678,6 +1696,26 @@ private void handleException(Exception cause) throws IOException {
             }
         }
 
+        if (cause instanceof SocketException) {
+            if (!conContext.isInboundClosed()) {
+                try {
+                    decode(null);
```

Comment:
One decode may be not sufficient if there are multiple pending handshake messages record, or the connection has been established and application data transportation is in progress.  I'm not sure if an additional decode() here really solve the problem as described in JDK-8259516.

I did not follow the issue of JDK-8259516.  Why an SSLException is desired but not SocketException, as if the socket has been closed?

Reply:
I have removed issues for JDK-8259516.